src/conclusion_stage_image.py

```diff
@@ -0,0 +1,68 @@
+"""
+Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You
+may not use this file except in compliance with the License. A copy of
+the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ANY KIND, either express or implied. See the License for the specific
+language governing permissions and limitations under the License.
+"""
+
+from image import DockerImage
+from context import Context
+from utils import generate_safety_report_for_image
+
+import os
+
+class ConclusionStageImage(DockerImage):
+    """
+    Class designed to handle the ConclusionStageImages
+    """
+
+    def pre_build_configuration(self):
+        """
+        Conducts all the pre-build configurations from the parent class and then conducts
+        Safety Scan on the images generated in previous stage builds. The safety scan generates
+        the safety_report which is then baked into the image. 
+        """
+        ## Call the pre_build_configuration steps from the parent class
+        super(ConclusionStageImage, self).pre_build_configuration()
+        ## Generate safety scan report for the first stage image and add the file to artifacts
+        first_stage_image_uri = self.build_args['INITIAL_STAGE_IMAGE']
+        processed_image_uri = first_stage_image_uri.replace('.','-').replace('/','-').replace(':','-')
+        image_name = self.name
+        tarfile_name_for_context = f"{processed_image_uri}-{image_name}"
+        storage_file_path = f"{os.getenv('ROOT_FOLDER_PATH')}/src/{tarfile_name_for_context}_safety_report.json"
```

Comment:
use os.path.join on this one - also instead of calling `os.getenv('ROOT_FOLDER_PATH') everywhere, you can create a utility function and use CODEBUILD_SRC_DIR when available, and have some fallback option when not available. Example:

```
def get_root_folder_path():
    root_dir_pattern = re.compile(r"^(\S+deep-learning-containers)")
    pwd = os.getcwd()
    return os.getenv('CODEBUILD_SRC_DIR', root_dir_pattern.match(pwd).group(1))
```

Reply:
Thanks, will do. Follow up question: I can't find `CODEBUILD_SRC_DIR` in the list of already set env variables in the CodeBuild image

Reply:
Done